addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -0,0 +1,577 @@
+import org.openhab.binding.somfytahoma.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.openhab.binding.somfytahoma.SomfyTahomaBindingConstants.*;
+
+/**
+ * The {@link SomfyTahomaBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+public class SomfyTahomaBridgeHandler extends ConfigStatusBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyTahomaBridgeHandler.class);
+
+    private String cookie;
+
+    /**
+     * Future to poll for updated
+     */
+    private ScheduledFuture<?> pollFuture;
+
+    /**
+     * Our configuration
+     */
+    protected SomfyTahomaConfig thingConfig;
+
+    //Gson & parser
+    private final Gson gson = new Gson();
+    //private final JsonParser parser = new JsonParser();
+    private SomfyTahomaItemDiscoveryService discoveryService = null;
+
+    public SomfyTahomaBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(VERSION)) {
+            // TODO: handle command
+
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        String thingUid = getThing().getUID().toString();
+        thingConfig = getConfigAs(SomfyTahomaConfig.class);
+        thingConfig.setThingUid(thingUid);
+
+        login();
+        scheduler.schedule(() -> startDiscovery(), 1, TimeUnit.SECONDS);
+        initPolling(thingConfig.getRefresh());
+    }
+
+    /**
+     * starts this things polling future
+     */
+    private void initPolling(int refresh) {
+        stopPolling();
+        pollFuture = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    updateTahomaStates();
+                } catch (Exception e) {
+                    logger.debug("Exception during poll : {}", e);
+                }
+            }
+        }, 10, refresh, TimeUnit.SECONDS);
+
+    }
+
+    private void login() {
+        String url = null;
+
+        if ((thingConfig.getEmail() != null && thingConfig.getEmail().isEmpty()) || (thingConfig.getPassword() != null && thingConfig.getPassword().isEmpty())) {
```

Comment:
You could check the email and password files with StringUtils.isEmpty (from the apache library). Would make those comparisions a bit nicer to read 😄